collect_app/src/main/java/org/odk/collect/android/widgets/utilities/FileWidgetUtils.java

```diff
@@ -1,19 +1,68 @@
 package org.odk.collect.android.widgets.utilities;
 
+import android.content.ContentValues;
+import android.content.Context;
+import android.net.Uri;
+import android.provider.MediaStore;
+
 import androidx.annotation.NonNull;
 
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.javarosawrapper.FormController;
 import org.odk.collect.android.utilities.FileUtil;
+import org.odk.collect.android.utilities.MediaUtil;
 
 import java.io.File;
 
 public class FileWidgetUtils {
 
     private FileWidgetUtils() {
+    }
+
+    public static File getFile(Context context, Uri fileUri) {
```

Comment:
I'm wondering if uses of this could be replaced with the new `FormSaveViewModel#createAnswerFile` (which was added on top of the work you did with `QuestionMediaManager` 😁)? I think instead of `FileWidgetUtils.getFile` it could be:

```java
File sourceFile = new File(getSourcePathFromUri(context, fileUri, MediaStore.MediaColumns.DATA));
setWidgetData(formSaveViewModel.createAnswerFile(sourceFile));
```

In the widget itself you can use `QuestionMediaManager#getAnswerFile` (like in `AudioWidget`). 

What do you think?

Reply:
Yes I was also thinking to get rid of `FileWidgetUtils.getFile()` so that everything's consistent.